### ISSUE TITLE
Passive -> Leader transaction propagation is broken after leader restart

### ISSUE THREAD

**Issue by dkijania** [input-output-hk/jormungandr#2311]
**Describe the bug**
In simple topology:  Passive Node -> Leader Node an error (?) occurs: if leader is restarted then its having problems with picking up fragments from passive node. Policy settings for both nodes:
```
  policy:
    quarantine_duration: 30s
``` 

**Mandatory Information**
1. `jcli --full-version` newest master;
2. `jormungandr --full-version` newest master;

**To Reproduce**
1. Start leader node 
2. Start passive node
3. Give some time to produce blocks by leader node.
4. Send first fragment to passive node - should be in block in both nodes fragment logs.
5. Stop leader node.
6. Send second fragment to passive - it should be pending in passive node.
7. Start leader node again on old storage (same command as at step 1.) 
8. Wait more than 30 s (quarantine setting)
9. Send third fragment to passive

**Actual behavior**
Passive node keeps Leader node in quarantine all the time. Only first fragment are in block for both nodes, fragment 2 and 3 are in pending state for passive node, leader node has empty fragment logs after restart

**Expected behavior**
first and third fragment should be in block in leader node fragment logs. Node should reestablish connection.

**Additional context**
In order to reproduce issue please run test `leader_restart` from jormungandr-scenario-tests:
```
cd jormungandr-scenario-tests
cargo run -- --scenario --name leader_restart
```

or use given package to setup network:
[leader_restart.zip](https://github.com/input-output-hk/jormungandr/files/4722505/leader_restart.zip)

**Comment by eugene-babichenko** [comment]
Confirming that this is a quarantine issue.

**Comment by eugene-babichenko** [comment]
An update on this: `Policy::check` in `poldercast` is never triggered after sending a node to quarantine, although `poldercast` is notified about the incoming gossips from the quarantined node.

**Comment by dkijania** [comment]
Issue is fixed in 0.13.0 version of jormungandr
